applications/RomApplication/python_scripts/rom_database.py

```diff
@@ -0,0 +1,330 @@
+                        (tol_sol, tol_res, projection_type, file_name))
+        elif table_name == 'Neural_Network':
+            cursor.execute(f'INSERT INTO {table_name}  (tol_sol , modes , layers_size, batch_size, epochs, scheduler, base_lr, additional_params, model_number, file_name) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)',
+                        (tol_sol, nn_data6_str, nn_data7_str, nn_data8_int, nn_data9_int, nn_data10_str, nn_data11_double, nn_data12_str, nn_data13_int, file_name))
+        else:
+            err_msg = f'Error: table_name: {table_name} not available. Available options are: {", ".join(self.table_names)}'
+            raise Exception(err_msg)
+
+        conn.commit()
+        conn.close()
+
+        if table_name == "Neural_Network":
+            pass
+        else:
+            self.save_as_npy(numpy_array, file_name)
+
+
+
+    def identify_list_type(self, lst):
+        if not lst:
+            return "Empty list"
+        if isinstance(lst[0], list):
+            if any(isinstance(subitem, list) for item in lst for subitem in item):
+                return "Invalid structure - nested lists found"
+            return "complete_mu"
+        else:
+            if any(isinstance(item, list) for item in lst):
+                return "Invalid structure - mixed types"
+            return "mu"
+
+
+
+    def serialize_mu(self, parameters):
+        return json.dumps(parameters)
+
+
+    def save_as_npy(self, result, hash_mu):
+        file_path = self.npys_directory / f"{hash_mu}.npy"
+        np.save(file_path, result)
+        print(f"numpy saved to {file_path}")
+
+
+    def make_mu_dictionary(self, mu):
+        if self.mu_names is None:
+            self.mu_names = []
+            for i in range(len(mu)):
+                self.mu_names.append(f'generic_name_{i}')
+        return dict(zip(self.mu_names , mu))
+
+
+    def get_snapshots_matrix_from_database(self, mu_list, table_name='FOM_Fit'):
+        if mu_list == [None]: #this happens when no mu is passed, the simulation run is the one in the ProjectParameters.json
+            mu_list_unique = mu_list
+        else:
+            unique_tuples = set(tuple(item) for item in mu_list)
+            mu_list_unique = [list(item) for item in unique_tuples] #unique members in mu_lust
+        SnapshotsMatrix = []
+        conn = sqlite3.connect(self.database_name)
+        cursor = conn.cursor()
+        for mu in mu_list_unique:
+            hash_mu, _ = self.get_hashed_mu_for_table(table_name, mu)
+            cursor.execute(f"SELECT file_name FROM {table_name} WHERE file_name = ?", (hash_mu,))
+            result = cursor.fetchone()
+            if result:
+                SnapshotsMatrix.append(self.get_single_numpy_from_database(result[0]))
+            else:
+                print(f"No entry found for hash {hash_mu}")
+        conn.close()
+        return np.block(SnapshotsMatrix) if SnapshotsMatrix else None
```

Comment:
Maybe it should also return a bool stating the mu values that weren't found in the database. So that one may alert the user that the matrix isn't complete, and ask them to simulate the remaining ones.

Reply:
Added warning, but could not trigger that behaviour. The cases are check to exist both in Fit() and Test() in the RomManger. It will serve for debugging when adding more tables